src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/base.py

```diff
@@ -113,7 +119,7 @@ def _validate_operational_conditions(
                 np.where(np.any(rate < 0, axis=0), 0, rate) if np.ndim(rate) == 2 else np.where(rate < 0, 0, rate)
             )  # if the rate for one stream is negative, set the rates for all streams to zero for that time step
         if intermediate_pressure is not None:
-            if not np.all(intermediate_pressure > 0):
+            if not np.all(intermediate_pressure[check_validity_for_time_step] > 0):
```

Comment:
How is `check_validity_for_time_step` used here? Not sure I understand what is happening here